app2/src/main/java/com/example/app2/service/OrderService.java

```diff
@@ -0,0 +1,66 @@
+package com.example.app2.service;
+
+import com.example.app1.pojo.Order;
+import com.example.app2.pojo.Summary;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jms.annotation.JmsListener;
+import org.springframework.jms.listener.adapter.JmsResponse;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.handler.annotation.Payload;
+import org.springframework.stereotype.Service;
+
+@Service
+public class OrderService {
+    public static final Logger LOGGER = LoggerFactory.getLogger(OrderService.class);
+
+    @Autowired
+    private OrderProcessingService orderProcessingService;
+
+    @Autowired
+    Summary summary;
+
+    @JmsListener(destination = "orders", selector = "GOODS_TYPE = 'ITEMS'")
+    public JmsResponse<Message<Summary>> handleItem(@Payload Order order) {
+        JmsResponse<Message<Summary>> response = null;
+        Summary backup = orderProcessingService.backupSummary();
+        try {
+            response = orderProcessingService.processItemsMessage(order);
+        } catch (Exception e) {
+            LOGGER.warn(e + ": " + e.getMessage() + ". " + e.getCause());
```

Comment:
maybe logic inside 'catch' can be moved to separate private method. it is quite common for methods within this class:
```
LOGGER.warn(e + ": " + e.getMessage() + ". " + e.getCause());
summary = backup;
throw new RuntimeException("Exception during handling item", e);
```